src/data-loader.ts

```
@@ -1452,12 +1450,30 @@ export async function calculateContextTokens(transcriptPath: string): Promise<{
 						+ (usage.cache_creation_input_tokens ?? 0)
 						+ (usage.cache_read_input_tokens ?? 0);
 
-				const percentage = Math.min(100, Math.max(0, Math.round((inputTokens / CONTEXT_LIMIT) * 100)));
+				// Get context limit from PricingFetcher
+				let contextLimit = 200_000; // Fallback for when modelId is not provided
+				if (modelId != null && modelId !== '') {
+					using fetcher = new PricingFetcher(offline);
+					const contextLimitResult = await fetcher.getModelContextLimit(modelId);
```

ESLint

[error] Unsafe assignment of an error typed value.

(ts/no-unsafe-assignment)